File: backend/CarRentalPlatformBackend/src/main/java/com/poly/carrentalplatformbackend/services/MaintenanceService.java
package com.poly.carrentalplatformbackend.services;

import com.poly.carrentalplatformbackend.entities.Maintenance;

import java.util.List;

public interface MaintenanceService {



    Maintenance getMaintenance(int id);

    List<Maintenance> getAllMaintenances();

    List<Maintenance> getMaintenancesByVoiture(int voitureId);

    Maintenance createMaintenance(Maintenance maintenance);

    Maintenance terminerMaintenance(int id);

    void deleteMaintenance(int id);
}

File: frontend/car-rental-frontend/src/api/maintenance.service.js
import api from "./axios";

// ✅ READ ALL (ADMIN)
export const getAllMaintenances = () => api.get("/maintenances");

// ✅ READ ONE (ADMIN)
export const getMaintenanceById = (id) => api.get(`/maintenances/${id}`);

// ✅ READ BY VOITURE (ADMIN)
export const getMaintenancesByVoiture = (voitureId) =>
  api.get(`/maintenances/voiture/${voitureId}`);

// ✅ CREATE (ADMIN)
export const createMaintenance = (data) => api.post("/maintenances", data);

// ✅ TERMINER (ADMIN)
export const terminerMaintenance = (id) => api.put(`/maintenances/${id}/terminer`);

// ✅ DELETE (ADMIN)
export const deleteMaintenance = (id) => api.delete(`/maintenances/${id}`);
